programs/vote/src/vote_state/mod.rs

```diff
@@ -967,6 +1134,33 @@ pub fn process_vote<S: std::hash::BuildHasher>(
     vote_account.set_state(&VoteStateVersions::new_current(vote_state))
 }
 
+pub fn process_vote_state_update<S: std::hash::BuildHasher>(
+    vote_account: &KeyedAccount,
+    slot_hashes: &[SlotHash],
+    clock: &Clock,
+    vote_state_update: &VoteStateUpdate,
+    signers: &HashSet<Pubkey, S>,
+) -> Result<(), InstructionError> {
+    let versioned = State::<VoteStateVersions>::state(vote_account)?;
+
+    if versioned.is_uninitialized() {
+        return Err(InstructionError::UninitializedAccount);
+    }
+
+    let mut vote_state = versioned.convert_to_current();
+    let authorized_voter = vote_state.get_and_update_authorized_voter(clock.epoch)?;
+    verify_authorized_signer(&authorized_voter, signers)?;
```

Comment:
this is the same as the checks made in `fn process_vote()`, maybe can be extracted into a common utility?